dff/messengers/telegram/abstract.py

```diff
@@ -0,0 +1,631 @@
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        if media_group_attachments_num != 1:
+                            files += [
+                                InputMediaAnimation(
+                                    attachment_bytes,
+                                    **generate_extra_fields(
+                                        attachment, ["filename", "caption", "parse_mode", "has_spoiler", "thumbnail"]
+                                    ),
+                                ),
+                            ]
+                        else:
+                            await bot.send_animation(
+                                chat_id,
+                                attachment_bytes,
+                                caption=message.text,
+                                **generate_extra_fields(
+                                    attachment,
+                                    [
+                                        "parse_mode",
+                                        "disable_notification",
+                                        "reply_markup",
+                                        "has_spoiler",
+                                        "thumbnail",
+                                        "filename",
+                                        "message_effect_id",
+                                    ],
+                                ),
+                            )
+                            message_text_covered = True
+                if isinstance(attachment, Image):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        if media_group_attachments_num != 1:
+                            files += [
+                                InputMediaPhoto(
+                                    attachment_bytes,
+                                    **generate_extra_fields(
+                                        attachment, ["filename", "caption", "parse_mode", "has_spoiler"]
+                                    ),
+                                ),
+                            ]
+                        else:
+                            await bot.send_photo(
+                                chat_id,
+                                attachment_bytes,
+                                caption=message.text,
+                                **generate_extra_fields(
+                                    attachment,
+                                    [
+                                        "disable_notification",
+                                        "reply_markup",
+                                        "parse_mode",
+                                        "has_spoiler",
+                                        "filename",
+                                        "message_effect_id",
```

Comment:
I don't like how extra fields differ depending on if there are multiple attachments in the list.

My proposal:
1. Inherit `disable_notification`, `reply_markup`, `message_effect_id`, `protect_content` from the message object.
2. Always use `attachment.caption` as caption. Send `message.text` via `send_message` only. Remove `message_text_covered`.

Another option (an alternative to the first point) is to add a mediagroup attachment so that users can group attachments on their own.
That way it is much more clear in terms of the API:
```
Message(attachments=[Image("always sent separately"), MediaGroup(Image("sent in a group"), Video("sent in a group"))])
```

This makes the API more consistent (currently, if I put an attachment with `caption` in attachments then `caption` has no effect, but if I put two of those attachments in the list then `caption` starts to matter).